test/e2e/framework/conformance/conformance.go

```diff
@@ -0,0 +1,845 @@
+	err = restClient.
+		Get().
+		AbsPath(path).
+		Do(ctx).
+		Into(&api)
+	framework.ExpectNoError(err, "get resource API")
+	resources := api.Object["resources"].([]any)
+	index := slices.IndexFunc(resources, func(entry any) bool {
+		return entry.(map[string]any)["name"].(string) == resource
+	})
+	if index < 0 {
+		framework.Failf("API for %s does not include entry for %s, got:\n%s", gv, resource, marshalYAML(&api, 1))
+	}
+
+	// Create the initial resource.
+	ginkgo.By(fmt.Sprintf("Creating:\n%s", marshalYAML(desiredInitialObject, 1)))
+	existingObject, err := client.Create(ctx, desiredInitialObject, metav1.CreateOptions{FieldValidation: "Strict"})
+	framework.ExpectNoError(err, "create initial %s", resourceType)
+	ginkgo.DeferCleanup(func(ctx context.Context) {
+		// Always clean up.
+		err = client.Delete(ctx, desiredInitialObject.GetName(), metav1.DeleteOptions{})
+		if apierrors.IsNotFound(err) {
+			return
+		}
+		framework.ExpectNoError(err, "delete %s", resourceType)
+		ensureNotFound(ctx, getResource)
+	})
+	verifyObject("create", desiredInitialObject, existingObject)
+
+	// Get to check for existence.
+	ginkgo.By(fmt.Sprintf("Getting %s", klog.KObj(desiredInitialObject)))
+	existingObject, err = client.Get(ctx, desiredInitialObject.GetName(), metav1.GetOptions{})
+	framework.ExpectNoError(err, "get updated %s", resourceType)
+	verifyObject("get", desiredInitialObject, existingObject)
+
+	// Update the resource. Retry because the existing object might have been updated in the meantime.
+	mustGet := false
+	gomega.Eventually(ctx, func(ctx context.Context) error {
+		if mustGet {
+			ginkgo.By(fmt.Sprintf("Getting updated %s", klog.KObj(desiredInitialObject)))
+			existingObject, err = getResource(ctx)
+			if err != nil {
+				return fmt.Errorf("get existing %s: %w", resourceType, err)
+			}
+		}
+		object := tc.GetUpdateSpec(existingObject.DeepCopy())
+		ginkgo.By(fmt.Sprintf("Updating:\n%s", marshalYAML(object, 1)))
+		existingObject, err = client.Update(ctx, object, metav1.UpdateOptions{})
+		if err == nil {
+			return nil
+		}
+		mustGet = apierrors.IsConflict(err)
+		if mustGet {
+			// Retry immediately.
+			return fmt.Errorf("update %s: %w", resourceType, err)
+		}
+		if retry, retryAfter := framework.ShouldRetry(err); retry {
+			// Retry with a delay.
+			return gomega.TryAgainAfter(retryAfter)
+		}
+		// Give up, some other error occurred.
+		return gomega.StopTrying(fmt.Sprintf("update %s", resourceType)).Wrap(err)
+	}).Should(gomega.Succeed())
+	verifyObject("update", desiredUpdatedObject, existingObject)
+
+	// Same for the status. In addition, read the status (same result, but different endpoint+method).
+	if tc.HasStatus() {
+		mustGet := false
+		gomega.Eventually(ctx, func(ctx context.Context) error {
+			if mustGet {
+				ginkgo.By(fmt.Sprintf("Getting updated %s", klog.KObj(desiredInitialObject)))
+				existingObject, err = client.Get(ctx, desiredInitialObject.GetName(), metav1.GetOptions{})
```

Comment:
I think we always want to exercise a get of the status subresource to make sure that works